### PR TITLE
Update Neuron initializations

### PR DESCRIPTION
**Pull request by jeffhataws**
Update setting of NEURON_LIBRARY_PATH, Neuron environments for xmp.spawn/torchrun and other defaults.

### CODE REVIEW

File: torch_xla/__init__.py [hunk 1]
```diff
@@ -116,11 +116,27 @@ def _summarize_fn_tracker():
...
   try:
-    from torch_neuronx import xla
+    from libneuronxla.libneuronpjrt_path import libneuronpjrt_path
   except ImportError:
-    return
+    pass
   else:
-    xla.init()
+    os.environ.setdefault('PJRT_DEVICE', 'NEURON')
+    os.environ.setdefault('NEURON_LIBRARY_PATH', libneuronpjrt_path())
```
Comment (will-cromar) [inline torch_xla/__init__.py:124]: You can put the default path in the `NeuronPlugin` here: https://github.com/pytorch/xla/blob/f45775a5aeafb5219a045ab96d240b89c4106f42/torch_xla/_internal/neuron.py#L32

Long-term, you should put that plugin in your `libneuronxla` package and [register it as an `entry-point`](https://github.com/pytorch/xla/blob/master/docs/plugins.md#pytorchxla-plugin-package) so  you don't have to upstream Neuron-specific config changes. We can follow up on that after the release.

File: torch_xla/__init__.py [hunk 2]
```diff
@@ -116,11 +116,27 @@ def _summarize_fn_tracker():
...
 def _aws_ec2_inf_trn_init():
   try:
-    from torch_neuronx import xla
+    from libneuronxla.libneuronpjrt_path import libneuronpjrt_path
   except ImportError:
-    return
+    pass
   else:
-    xla.init()
+    os.environ.setdefault('PJRT_DEVICE', 'NEURON')
```
Comment (will-cromar) [inline torch_xla/__init__.py:123]: Can you move device detection to `_maybe_select_default_device`? 

https://github.com/pytorch/xla/blob/f45775a5aeafb5219a045ab96d240b89c4106f42/torch_xla/runtime.py#L56-L77

File: torch_xla/__init__.py [hunk 3]
```diff
@@ -116,11 +116,27 @@ def _summarize_fn_tracker():
...
+    try:
+      from torch_neuronx import xla
+    except ImportError:
+      pass
+    else:
+      xla.init()
+
+    # basic initializations if torch-neuronx is not available
+    from ._internal import neuron
+    if os.path.basename(sys.argv[0]) != 'neuron_parallel_compile':
```
Comment (will-cromar) [inline torch_xla/__init__.py:135]: Does this make sense in `configure_multiprocess`?

https://github.com/pytorch/xla/blob/f45775a5aeafb5219a045ab96d240b89c4106f42/torch_xla/experimental/plugins.py#L36-L43

If not, that's fine. After the release, I can add a new plugin method to be called during init (e.g. `on_register`).

File: torch_xla/_internal/neuron_utils.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        f"LOCAL_RANK (torchrun) or PJRT_LOCAL_PROCESS_RANK (xmp.spawn) value of {local_rank} is higher than "
+        + f"count {count} from NEURON_RT_VISIBLE_CORES {cores_list}")
+  else:
+    remapped_core = cores_list[local_rank]
+    logger.warning(f"Remapping NEURON_RT_VISIBLE_CORES {cores_list} to " +
+                   f"NEURON_RT_VISIBLE_CORES[LOCAL_RANK]={remapped_core}")
+    os.environ['NEURON_RT_VISIBLE_CORES'] = str(remapped_core)
+
+
+def test_get_visible_cores_list():
```
Comment (will-cromar) [inline torch_xla/_internal/neuron_utils.py:69]: Can you move `test_` methods to `test/` so they're not included in the `torch_xla` package?

Comment (jeffhataws) [reply torch_xla/_internal/neuron_utils.py:69]: @will-cromar do you recommend I add a test/neuron directory for Neuron specific tests?

Comment (will-cromar) [reply torch_xla/_internal/neuron_utils.py:69]: Yeah, that makes sense to me

File: torch_xla/__init__.py (hunk 2 above)
Comment (jeffhataws) [reply torch_xla/__init__.py]: done

File: torch_xla/__init__.py (hunk 1 above)
Comment (jeffhataws) [reply torch_xla/__init__.py:124]: I put the default path in [xla/torch_xla/_internal/neuron.py](https://github.com/pytorch/xla/blob/f45775a5aeafb5219a045ab96d240b89c4106f42/torch_xla/_internal/neuron.py#L32) also now, but I need NEURON_LIBRARY_PATH setting here for proper Neuron Cache operations.

File: torch_xla/__init__.py (hunk 3 above)
Comment (jeffhataws) [reply torch_xla/__init__.py:135]: For torchrun, this location is required. Unless you can show me another way.

**Review by will-cromar — APPROVED** [review]
LGTM

**Review by aws-kingrj — APPROVED** [review]

**Pull request merged**